server/src/apub/post.rs

```diff
@@ -51,6 +51,46 @@ impl ToApub for Post {
     let url = self.url.as_ref().filter(|u| !u.is_empty());
     if let Some(u) = url {
       oprops.set_url_xsd_any_uri(u.to_owned())?;
+
+      // Embeds
+      let mut page_preview = Page::new();
+      page_preview
+        .object_props
+        .set_url_xsd_any_uri(u.to_owned())?;
+
+      if let Some(embed_title) = &self.embed_title {
+        page_preview
+          .object_props
+          .set_name_xsd_string(embed_title.to_owned())?;
+      }
+
+      if let Some(embed_description) = &self.embed_description {
+        page_preview
+          .object_props
+          .set_summary_xsd_string(embed_description.to_owned())?;
+      }
+
+      if let Some(embed_html) = &self.embed_html {
+        page_preview
+          .object_props
+          .set_content_xsd_string(embed_html.to_owned())?;
+      }
+
+      oprops.set_preview_base_box(page_preview)?;
+    }
+
+    if let Some(thumbnail_url) = &self.thumbnail_url {
+      let full_url = format!(
+        "{}://{}/pictshare/{}",
+        get_apub_protocol_string(),
+        Settings::get().hostname,
+        thumbnail_url
+      );
+
+      let mut image = Image::new();
+      image.object_props.set_url_xsd_any_uri(full_url)?;
+      let any_image = AnyImage::from_concrete(image)?;
+      oprops.set_image_any_image(any_image)?;
```

Comment:
I think we should also set the image size, but I dont see any attributes for that in the library...

Reply:
In the activitypub docs, it didn't have that for general images, but it did have it for `icon`, which I didn't do yet. Mainly because its better to actually store the full URL, and then let front end requests to `/pictshare/image/96x96` or whichever do the sizing. I think for the inline comments and such, i use a smaller size than on the user page. 

We might have to add height and width for compatibility later, but lets worry about another time.